main.go

```diff
@@ -175,11 +181,35 @@ func handle(w http.ResponseWriter, r *http.Request) {
 		defer f.Close()
 		http.ServeContent(w, r, "main.wasm", time.Now(), f)
 		return
+
+	case "_wait":
+		waitForUpdate(w, r)
+		return
+	case "_notify":
+		notifyWaiters(w, r)
+		return
 	}
 
 	http.ServeFile(w, r, filepath.Join(".", r.URL.Path))
 }
 
+func waitForUpdate(w http.ResponseWriter, r *http.Request) {
+	waitChannel <- struct{}{}
+	http.ServeContent(w, r, "", time.Now(), bytes.NewReader(nil))
+}
+
+func notifyWaiters(w http.ResponseWriter, r *http.Request) {
+	for {
```

Comment:
I think we don't need this infinite loop.

Reply:
Ah sorry, this is needed.

Reply:
If nothing is waited, `_notify` should return immediately. Then the below code sounds enough.

```
select {
case <-waitChannel:
default:
}
http.ServeContent(w, r, "", time.Now(), bytes.NewReader(nil))
```

Reply:
We need the loop for the case when there are multiple browsers waiting - we need to clear all "messages" sent on the channel.

Reply:
This sounds like Java's `notifyAll` rather than `notify`, but I understand the necessity.